mayastor/src/bdev/nvmx/controller.rs

```diff
@@ -821,22 +822,36 @@ extern "C" fn aer_cb(ctx: *mut c_void, cpl: *const spdk_nvme_cpl) {
     }
 }
 
-/// return number of completions processed (maybe 0) or the negated on error,
-/// which is one of:
-///
-/// ENXIO: the qpair is not conected  or when the controller is
-/// marked as failed.
-///
-/// EGAIN: returned whenever the controller is being reset.
+/// Poll to process qpair completions on admin queue
+/// Returns: 0 (SPDK_POLLER_IDLE) or 1 (SPDK_POLLER_BUSY)
 pub extern "C" fn nvme_poll_adminq(ctx: *mut c_void) -> i32 {
     let mut context = NonNull::<TimeoutConfig>::new(ctx.cast())
         .expect("ctx pointer may never be null");
     let context = unsafe { context.as_mut() };
 
+    // returns number of completions processed (maybe 0) or the negated error,
+    // which is one of:
+    //
+    // ENXIO: the qpair is not connected or when the controller is
+    // marked as failed.
+    //
+    // EAGAIN: returned whenever the controller is being reset.
     let result = context.process_adminq();
 
     if result < 0 {
-        //error!("{}: {}", context.name, Errno::from_i32(result.abs()));
+        error!(
```

Comment:
Don't we see lots of errors flooding the console ? I do recall some situations where logs were full of error messages related to failures upon processing adminq completions, that's why it was commented out.

Reply:
Yes, without scheduling the device destroy future the log will flood. With it, this is seen only once on an idle nexus, maybe 3-5 times on a nexus with an IO load.

Reply:
Good then, thanks.